rita/src/rita_client/dashboard/nodeinfo.rs

```diff
@@ -0,0 +1,107 @@
+/*
+The nodeinfo endpoint gathers info about neighbors for the dashbaord
+*/
+
+use actix::prelude::*;
+use failure::Error;
+use futures::Future;
+use serde_json;
+use std::net::{SocketAddr, TcpStream};
+
+use babel_monitor::Babel;
+use num256::Int256;
+use rita_common::dashboard::Dashboard;
+use rita_common::debt_keeper::{DebtKeeper, Dump};
+use settings::RitaClientSettings;
+use settings::RitaCommonSettings;
+use SETTING;
+
+#[derive(Serialize)]
+pub struct NodeInfo {
+    pub nickname: String,
+    pub route_metric_to_exit: u16,
+    pub total_payments: Int256,
+    pub debt: i64,
+    pub link_cost: u16,
+    pub price_to_exit: u32,
+}
+
+pub struct GetNodeInfo;
+
+impl Message for GetNodeInfo {
+    type Result = Result<Vec<NodeInfo>, Error>;
+}
+
+impl Handler<GetNodeInfo> for Dashboard {
+    type Result = ResponseFuture<Vec<NodeInfo>, Error>;
+
+    fn handle(&mut self, _msg: GetNodeInfo, _ctx: &mut Self::Context) -> Self::Result {
+        Box::new(
+            DebtKeeper::from_registry()
+                .send(Dump {})
+                .from_err()
+                .and_then(|res| {
+                    let stream = TcpStream::connect::<SocketAddr>(
+                        format!("[::1]:{}", SETTING.get_network().babel_port).parse()?,
+                    )?;
+                    let mut babel = Babel::new(stream);
+                    babel.start_connection()?;
+                    let route_table_sample = babel.parse_routes()?;
+
+                    let res = res?;
```

Comment:
I'd consier doing `res?` at the beggining of the function to return early with potential error. This would save on doing TCP connection, and babel instance creation.

Reply:
good point.